### DISCUSSION TITLE
add reducedDims to ArchR project from Seurat clustering

### DISCUSSION THREAD

**Discussion by mitchTwoTimes** [GreenleafLab/ArchR#1377] [Questions / Documentation]
Hello,

Is there a way to add Seurat clustering results to the `reducedDims` slot of an ArchR project? Here is the dimensional reduction analysis I would like to add:
```
> str(seurat.rna@reductions$global_umap)
Formal class 'DimReduc' [package "SeuratObject"] with 9 slots
  ..@ cell.embeddings           : num [1:65078, 1:2] -8.98 -11.48 -11.62 -8.31 -8.47 ...
  .. ..- attr(*, "scaled:center")= num [1:2] -0.0731 -0.0396
  .. ..- attr(*, "dimnames")=List of 2
  .. .. ..$ : chr [1:65078] "d7_br1#ACCTTCATCTCTAGCC-1" "d7_br1#GGTTGCTCAATGCGCT-1" "d7_br1#CTGGTAAGTAGTTGGC-1" "d7_br1#TGCTTAAAGGAATAAC-1" ...
  .. .. ..$ : chr [1:2] "UMAP_1" "UMAP_2"
  ..@ feature.loadings          : num[0 , 0 ] 
  ..@ feature.loadings.projected: num[0 , 0 ] 
  ..@ assay.used                : chr "originalexp"
  ..@ global                    : logi TRUE
  ..@ stdev                     : num(0) 
  ..@ key                       : chr "UMAP_"
  ..@ jackstraw                 :Formal class 'JackStrawData' [package "SeuratObject"] with 4 slots
  .. .. ..@ empirical.p.values     : num[0 , 0 ] 
  .. .. ..@ fake.reduction.scores  : num[0 , 0 ] 
  .. .. ..@ empirical.p.values.full: num[0 , 0 ] 
  .. .. ..@ overall.p.values       : num[0 , 0 ] 
  ..@ misc                      : list()
```

Thank you

**Comment by rcorces** [comment]
There isnt a function to do this but here is some example code that might help:

```
colData2 <- readRDS("oldMetaData.rds")
all.equal(rownames(colData2), proj2$cellNames)

df <- DataFrame(row.names=proj2$cellNames, "custom#UMAP1" = colData2$UMAP1, "custom#UMAP2" =  colData2$UMAP2, check.names = FALSE)
proj2@embeddings$customUMAP <- SimpleList(df = df, params = list())
proj2@cellColData$Clusters <- colData2$Clusters
plotEmbedding(ArchRProj = proj2, embedding = "customUMAP", colorBy = "cellColData", name = "Clusters")

```

**Reply by mitchTwoTimes** [reply]
Thanks for the response. I was able to add my Seurat-generated UMAP coordinates to the `embeddings` slot and successfully recreate the plot. 

However, I actually need to add my Seurat UMAP info to the `reducedDims` slot to run `addPeak2GeneLinks`. The `reducedDims` slot seems a bit different than `embeddings` - any hints to deal with this?

**Reply by rcorces** [reply]
ah. my mistake. there are a lot more moving parts with `reducedDims`. I dont have code to do that and I dont off the top of my head know which components of the reducedDims slot is used in various parts of ArchR. You can try to mimic what its supposed to look like and see if it works.

**Reply by mitchTwoTimes** [reply]
Hey @rcorces, I wanted to revisit this. I ended up just using the LSI clustering to get a p2g-friendly reducedDims embedding. I colored by my Seurat clusters and they were basically recapitulated in the LSI space.

Would it be catastrophic if ran `addPeak2GeneLinks` on the LSI space but `groupBy` Seurat clusters?

**Reply by rcorces** [reply]
that sounds like it should be fine to me
